apps/kg/train_pytorch.py

```diff
@@ -147,3 +214,128 @@ def train_mp(args, model, train_sampler, valid_samplers=None, rank=0, rel_parts=
 @thread_wrapped_func
 def test_mp(args, model, test_samplers, rank=0, mode='Test', queue=None):
     test(args, model, test_samplers, rank, mode, queue)
+
+@thread_wrapped_func
+def dist_train_test(args, model, train_sampler, entity_pb, relation_pb, l2g, rank=0, rel_parts=None, cross_rels=None, barrier=None):
+    if args.num_proc > 1:
+        th.set_num_threads(args.num_thread)
+
+    client = connect_to_kvstore(args, entity_pb, relation_pb, l2g)
+    client.barrier()
+    train_time_start = time.time()
+    train(args, model, train_sampler, None, rank, rel_parts, cross_rels, barrier, client)
+    print('Total train time {:.3f} seconds'.format(time.time() - train_time_start))
```

Comment:
It is unfair to count train time without counting the barrier time.

Reply:
You are right.